pkg/kube/pod_runner_test.go

```diff
@@ -109,9 +114,88 @@ func (s *PodRunnerTestSuite) TestPodRunnerForSuccessCase(c *C) {
+func (s *PodRunnerTestSuite) TestPodRunnerWithDebugLabelForSuccessCase(c *C) {
+	jobIDSuffix := "JobID"
+	for _, tc := range []struct {
+		name               string
+		targetKey          string
+		contextKey         string
+		contextValue       string
+		expectedLabel      bool
+		validationErrorMsg string
+	}{
+		{
+			name:               "target key (kanister.io/JobID) present in pod labels",
+			targetKey:          path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextKey:         path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextValue:       "xyz123",
+			expectedLabel:      true,
+			validationErrorMsg: "Expected label to be set",
+		},
+		{
+			name:               "target key (kanister.io/JobID) not present in pod labels",
+			targetKey:          path.Join(consts.LabelPrefix, jobIDSuffix),
+			contextKey:         path.Join(consts.LabelPrefix, "NonJobID"),
+			contextValue:       "some-other-value",
+			expectedLabel:      false,
+			validationErrorMsg: "Expected label to be not set",
+		},
+	} {
+		ctx, cancel := context.WithCancel(context.Background())
+		ctx = field.Context(ctx, tc.contextKey, tc.contextValue)
+		cli := fake.NewSimpleClientset()
+		cli.PrependReactor("create", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			return false, nil, nil
+		})
+		cli.PrependReactor("get", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			p := &corev1.Pod{
+				Status: corev1.PodStatus{
+					Phase: corev1.PodRunning,
+				},
+			}
+			return true, p, nil
+		})
+		po := &PodOptions{
+			Namespace: podRunnerNS,
+			Name:      podName,
+			Command:   []string{"sh", "-c", "tail -f /dev/null"},
+		}
+		deleted := make(chan struct{})
+		cli.PrependReactor("delete", "pods", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+			c.Log("Pod deleted due to Context Cancelled")
+			close(deleted)
+			return true, nil, nil
+		})
+		AddLabelsToPodOptionsFromContext(ctx, po, tc.targetKey)
+		pr := NewPodRunner(cli, po)
+		errorCh := make(chan error)
+		go func() {
+			_, err := pr.Run(ctx, afterPodRunTestKeyPresentFunc(tc.targetKey, tc.validationErrorMsg, tc.expectedLabel, deleted))
+			errorCh <- err
+		}()
+		deleted <- struct{}{}
+		c.Assert(<-errorCh, IsNil)
+		cancel()
+	}
+}
+
 func makePodRunnerTestFunc(ch chan struct{}) func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
 	return func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
 		<-ch
 		return nil, nil
 	}
 }
+
+func afterPodRunTestKeyPresentFunc(labelKey, validationErrorMsg string, expectedLabel bool, ch chan struct{}) func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
+	return func(ctx context.Context, pc PodController) (map[string]interface{}, error) {
+		<-ch
+
+		_, got := pc.Pod().Labels[labelKey]
+		if got != expectedLabel {
+			return nil, errors.New(validationErrorMsg)
```

Comment:
```suggestion
			return nil, errors.New("Got different label than expected")
```
We don't need to declare `validationErrorMsg` per test case. We can use a generic error message.